python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -847,53 +845,49 @@ def resolve_automationCondition(
             return GrapheneAutomationCondition(self._asset_node_snap.automation_condition)
         return None
 
-    def _sensor_targets_asset(
-        self, sensor: RemoteSensor, asset_graph: RemoteAssetGraph, job_names: Set[str]
-    ) -> bool:
-        asset_key = self._asset_node_snap.asset_key
-
-        if sensor.asset_selection is not None:
-            try:
-                asset_selection = sensor.asset_selection.resolve(asset_graph)
-            except DagsterError:
-                return False
-
-            if asset_key in asset_selection:
-                return True
-
-        return any(target.job_name in job_names for target in sensor.get_targets())
-
     def resolve_targetingInstigators(self, graphene_info: ResolveInfo) -> Sequence[GrapheneSensor]:
-        repo = graphene_info.context.get_repository(self._repository_selector)
-        sensors = repo.get_sensors()
-        schedules = repo.get_schedules()
-
-        asset_graph = repo.asset_graph
-
-        job_names = {
-            job_name
-            for job_name in self._asset_node_snap.job_names
-            if not job_name == IMPLICIT_ASSET_JOB_NAME
-        }
+        if isinstance(self._remote_node, RemoteGlobalAssetNode):
+            # global nodes have saved references to their targeting instigators
+            schedules = [
+                graphene_info.context.get_schedule(schedule_handle)
+                for schedule_handle in self._remote_node.get_targeting_schedule_handles()
+            ]
+            sensors = [
+                graphene_info.context.get_sensor(sensor_handle)
+                for sensor_handle in self._remote_node.get_targeting_sensor_handles()
+            ]
+        else:
```

Comment:
gnarly but understandable given the constraints

we've got a test that hits both paths? will it be easy to accidentally fall back to the slow path if we're not careful - dunno if its worth trying to inject some kind of warning, or have the callsites that instantiate the node indicate that they're not expecting to need the repository or something (I guess that is determined already by what node you pass in)

Reply:
the conditions in which we resolve to global versus repository scoped asset nodes in the graphql layer is complicated and i have not tried to chase down a full understanding of when it occurs in test or production 

the meat of both paths is `get_schedules_targeting` either at resolver time here or at graph construction time in the global node case which gives me some belief things are working as expected